src/vector.c

```diff
@@ -171,6 +196,9 @@ vector_in(PG_FUNCTION_ARGS)
 		x[dim] = strtof(pt, &stringEnd);
 		CheckElement(x[dim]);
 		dim++;
+		// stringEnd is space is OK to support SELECT '[   1  ,       2     ]'::vector(2);
+		while (array_isspace((unsigned char)*stringEnd))
```

Comment:
No need to cast here or below

Reply:
seems we need that, this is handle `[1   , 2    ]` which has spaces between `1 and ,`

Reply:
`(unsigned char)` shouldn't be needed. Also, I think it should come after the `if (stringEnd == pt)` check.

Reply:
copy that, thanks for the info~